server/src/cm_cmd_exec.cpp

```diff
@@ -969,3 +752,515 @@ _size_to_byte_by_unit (double orgin_num, char unit)
+	nv_add_nvp(res, "location", volumes[i].location);
+	nv_add_nvp_int(res, "totalpage", volumes[i].total_size);
+	nv_add_nvp_int(res, "freepage", volumes[i].free_size);
+	ts_add_nvp_time(res, "date", volumes[i].date, "%04d%02d%02d",
+			NV_ADD_DATE);
+	nv_add_nvp(res, "close", "spaceinfo");
+    }
+
+    for (int i = 0; i < temporary_volumes.size(); i++) {
+	nv_add_nvp(res, "open", "spaceinfo");
+	nv_add_nvp(res, "spacename", temporary_volumes[i].vol_name);
+	nv_add_nvp(res, "type", temporary_volumes[i].purpose);
+	nv_add_nvp(res, "location", temporary_volumes[i].location);
+	nv_add_nvp_int(res, "totalpage", temporary_volumes[i].total_size);
+	nv_add_nvp_int(res, "freepage", temporary_volumes[i].free_size);
+	ts_add_nvp_time(res, "date", temporary_volumes[i].date, "%04d%02d%02d",
+			NV_ADD_DATE);
+	nv_add_nvp(res, "close", "spaceinfo");
+    }
+}
+
+void SpaceDbResultNewFormat::create_result(nvplist *res){
+    nv_update_val_int (res, "pagesize", page_size);
+    nv_update_val_int (res, "logpagesize", log_page_size);
+
+    for(int i = 0; i < 3; i++) {
+	nv_add_nvp(res, "open", "dbinfo");
+	nv_add_nvp(res, "type", databaseSpaceDescriptions[i].type);
+	nv_add_nvp(res, "purpose", databaseSpaceDescriptions[i].purpose);
+	nv_add_nvp_int(res, "volume_count", databaseSpaceDescriptions[i].volume_count);
+	nv_add_nvp_int(res, "used_size", databaseSpaceDescriptions[i].used_size);
+	nv_add_nvp_int(res, "free_size", databaseSpaceDescriptions[i].free_size);
+	nv_add_nvp_int(res, "total_size", databaseSpaceDescriptions[i].total_size);
+	nv_add_nvp(res, "close", "dbinfo");
+    }
+
+    for(int i = 0; i < volumes.size(); i++) {
+	nv_add_nvp(res, "open", "volumeinfo");
+	nv_add_nvp(res, "type", volumes[i].type);
+	nv_add_nvp(res, "purpose", volumes[i].purpose);
+	nv_add_nvp(res, "volume_name", volumes[i].volume_name);
+	nv_add_nvp_int(res, "volid", volumes[i].volid);
+	nv_add_nvp_int(res, "used_size", volumes[i].used_size);
+	nv_add_nvp_int(res, "free_size", volumes[i].free_size);
+	nv_add_nvp_int(res, "total_size", volumes[i].total_size);
+	nv_add_nvp(res, "close", "volumeinfo");
+    }
+
+    for(int i = 0; i < 4; i++){
+	nv_add_nvp(res, "open", "fileinfo");
+	nv_add_nvp(res, "data_type", fileSpaceDescriptions[i].data_type);
+	nv_add_nvp_int(res, "file_count", fileSpaceDescriptions[i].file_count);
+	nv_add_nvp_int(res, "used_size", fileSpaceDescriptions[i].used_size);
+	nv_add_nvp_int(res, "file_table_size", fileSpaceDescriptions[i].file_table_size);
+	nv_add_nvp_int(res, "reserved_size", fileSpaceDescriptions[i].reserved_size);
+	nv_add_nvp_int(res, "total_size", fileSpaceDescriptions[i].total_size);
+	nv_add_nvp(res, "close", "fileinfo");
+    }
+}
+
+int SpaceDbResultOldFormat::get_no_tpage(){
+    int no_tpage = 0, i;
+
+    for (i = 0; i < volumes.size(); i++) {
+	no_tpage += volumes[i].total_size;
+    }
+    for (i = 0; i < temporary_volumes.size(); i++) {
+	no_tpage += temporary_volumes[i].total_size;
+    }
```

Comment:
I think a return is missing.
I think that cnt_tpage is a better name than no_tpage.